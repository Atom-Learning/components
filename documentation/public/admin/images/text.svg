<svg width="32" height="13" viewBox="0 0 32 13" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M0.772727 1.61364V0.363636H9.5V1.61364H5.84091V12H4.43182V1.61364H0.772727ZM13.8864 12.1818C13.0455 12.1818 12.3201 11.9962 11.7102 11.625C11.1042 11.25 10.6364 10.7273 10.3068 10.0568C9.98106 9.38258 9.81818 8.59848 9.81818 7.70455C9.81818 6.81061 9.98106 6.02273 10.3068 5.34091C10.6364 4.6553 11.0947 4.12121 11.6818 3.73864C12.2727 3.35227 12.9621 3.15909 13.75 3.15909C14.2045 3.15909 14.6534 3.23485 15.0966 3.38636C15.5398 3.53788 15.9432 3.78409 16.3068 4.125C16.6705 4.46212 16.9602 4.90909 17.1761 5.46591C17.392 6.02273 17.5 6.70833 17.5 7.52273V8.09091H10.7727V6.93182H16.1364C16.1364 6.43939 16.0379 6 15.8409 5.61364C15.6477 5.22727 15.3712 4.92235 15.0114 4.69886C14.6553 4.47538 14.2348 4.36364 13.75 4.36364C13.2159 4.36364 12.7538 4.49621 12.3636 4.76136C11.9773 5.02273 11.6799 5.36364 11.4716 5.78409C11.2633 6.20455 11.1591 6.6553 11.1591 7.13636V7.90909C11.1591 8.56818 11.2727 9.12689 11.5 9.58523C11.7311 10.0398 12.0511 10.3864 12.4602 10.625C12.8693 10.8598 13.3447 10.9773 13.8864 10.9773C14.2386 10.9773 14.5568 10.928 14.8409 10.8295C15.1288 10.7273 15.3769 10.5758 15.5852 10.375C15.7936 10.1705 15.9545 9.91667 16.0682 9.61364L17.3636 9.97727C17.2273 10.4167 16.9981 10.803 16.6761 11.1364C16.3542 11.4659 15.9564 11.7235 15.483 11.9091C15.0095 12.0909 14.4773 12.1818 13.8864 12.1818ZM20.3054 3.27273L22.3963 6.84091L24.4872 3.27273H26.0327L23.2145 7.63636L26.0327 12H24.4872L22.3963 8.61364L20.3054 12H18.7599L21.5327 7.63636L18.7599 3.27273H20.3054ZM31.7415 3.27273V4.40909H27.2188V3.27273H31.7415ZM28.5369 1.18182H29.8778V9.5C29.8778 9.87879 29.9328 10.1629 30.0426 10.3523C30.1563 10.5379 30.3002 10.6629 30.4744 10.7273C30.6525 10.7879 30.84 10.8182 31.0369 10.8182C31.1847 10.8182 31.3059 10.8106 31.4006 10.7955C31.4953 10.7765 31.571 10.7614 31.6278 10.75L31.9006 11.9545C31.8097 11.9886 31.6828 12.0227 31.5199 12.0568C31.357 12.0947 31.1506 12.1136 30.9006 12.1136C30.5218 12.1136 30.1506 12.0322 29.7869 11.8693C29.4271 11.7064 29.1278 11.4583 28.8892 11.125C28.6544 10.7917 28.5369 10.3712 28.5369 9.86364V1.18182Z" fill="black"/>
</svg>
