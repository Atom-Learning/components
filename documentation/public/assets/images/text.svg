<svg width="32" height="13" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M.77 1.61V.36H9.5v1.25H5.84V12h-1.4V1.61H.76ZM13.9 12.18a3.74 3.74 0 0 1-3.58-2.13 5.35 5.35 0 0 1-.5-2.35c0-.89.17-1.68.5-2.36a3.7 3.7 0 0 1 3.44-2.18 4 4 0 0 1 1.35.23 3.34 3.34 0 0 1 2.08 2.08c.21.55.32 1.24.32 2.05v.57h-6.73V6.93h5.37c0-.5-.1-.93-.3-1.32a2.23 2.23 0 0 0-2.09-1.25 2.47 2.47 0 0 0-2.28 1.42 3 3 0 0 0-.31 1.36v.77c0 .66.11 1.22.34 1.68a2.48 2.48 0 0 0 2.39 1.39c.35 0 .67-.05.95-.15a1.93 1.93 0 0 0 1.23-1.22l1.3.37a2.9 2.9 0 0 1-.7 1.16c-.32.33-.71.58-1.19.77-.47.18-1 .27-1.6.27Zm6.42-8.9 2.09 3.56 2.09-3.57h1.54l-2.82 4.37L26.03 12H24.5l-2.1-3.39L20.32 12h-1.55l2.77-4.36-2.77-4.37h1.55Zm11.43 0V4.4h-4.52V3.27h4.52Zm-3.2-2.1h1.34V9.5c0 .38.05.66.16.85.12.19.26.31.43.38.18.06.37.09.57.09.14 0 .27 0 .36-.02l.23-.05.27 1.2a2.73 2.73 0 0 1-1 .16 2.7 2.7 0 0 1-1.11-.24 2.21 2.21 0 0 1-.9-.74 2.14 2.14 0 0 1-.35-1.27V1.18Z" fill="#000"/></svg>