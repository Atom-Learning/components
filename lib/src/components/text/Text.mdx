---
title: Text
component: Text
description: Text is our basic component for rendering text, use it for any non-heading text
category: Content
---

## Sizes

[Capsize](https://seek-oss.github.io/capsize/) is used to size the text to the height of its capital letters, trimming the space above capital letters and below the baseline.

```tsx preview
<Text size="sm">
  Anything smaller than body text. E.g. "forgot your password?" prompts
</Text>
```

```tsx preview
<Text>Body text. Most text in the app will use this.</Text>
```

```tsx preview
<Text size="lg">Large, non-heading text. E.g. a drop quote in an article</Text>
```
