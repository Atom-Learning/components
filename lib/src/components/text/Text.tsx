import * as React from 'react'

import { CSS, styled } from '~/stitches'
import type { Override } from '~/utilities'
import { capsize } from '~/utilities'

const defaultSize = 'md'

export const textVariants = {
  size: {
    xs: { fontSize: '$xs', lineHeight: 1.6, ...capsize(0.4364) },
    sm: { fontSize: '$sm', lineHeight: 1.53, ...capsize(0.4056) },
    md: { fontSize: '$md', lineHeight: 1.5, ...capsize(0.3864) },
    lg: { fontSize: '$lg', lineHeight: 1.52, ...capsize(0.3983) },
    xl: { fontSize: '$xl', lineHeight: 1.42, ...capsize(0.3506) }
  },
  noCapsize: {
    true: {
      '&::before, &::after': { display: 'none' }
    }
  }
}

export const getTextVariant: (options: {
  size: keyof typeof textVariants.size
  noCapsize?: boolean
}) => CSS = ({ size, noCapsize }) => ({
  ...textVariants.size[size],
  ...textVariants.noCapsize[`${noCapsize}`]
})

export const StyledText = styled('p', {
  color: '$tonal600',
  fontFamily: '$body',
  fontWeight: 400,
  margin: 0,
  /** Allow nesting `<Text />` inside `<Text />` without forcing a new line and spacing. */
  '& > &': {
    '&:before, &:after': { display: 'none' }
  },
  variants: textVariants
})

type TextProps = Override<
  React.ComponentProps<typeof StyledText>,
  {
    as?:
      | 'blockquote'
      | 'caption'
      | 'dd'
      | 'dt'
      | 'figcaption'
      | 'li'
      | 'p'
      | 'span'
      | 'legend'
      | React.ComponentType
      | React.ElementType
  }
>

export const Text: React.ForwardRefExoticComponent<TextProps> =
  React.forwardRef(({ size = defaultSize, ...remainingProps }, ref) => (
    <StyledText size={size} {...remainingProps} ref={ref} />
  ))

Text.displayName = 'Text'
