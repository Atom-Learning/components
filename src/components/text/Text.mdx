---
title: Text
component: Text
description: Text is our basic text component, use it for any non-heading text.
category: Content
---

## Sizes

```jsx preview
<Text size="sm">
  Anything smaller than body text. E.g. "forgot your password?" prompts
</Text>
```

```jsx preview
<Text>Body text. Most text in the app will use this. </Text>
```

```jsx preview
<Text size="lg">Large, non-heading text. E.g. a drop quote in an article</Text>
```
