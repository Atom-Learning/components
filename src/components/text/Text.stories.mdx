import { Canvas, Meta, Story, ArgsTable } from '@storybook/addon-docs/blocks'

import { Text } from '.'

<Meta title="Primitives/Text" component={Text} />

# Text

`Text` is our basic text component. Use it for any non-heading text.

## Sizes

### `sm`

<Text size="sm">
  Anything smaller than body text. E.g. "forgot your password?" prompts
</Text>

### `md`

<Text> Body text. Most text in the app will use this. </Text>

### `lg`

<Text size="lg"> Large, non-heading text. E.g. a drop quote in an article</Text>
