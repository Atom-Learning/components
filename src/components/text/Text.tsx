import * as React from 'react'

import { styled } from '~/stitches'
import { capsize, Override } from '~/utilities'

export const textVariantSize = {
  sm: {
    fontSize: '$sm',
    lineHeight: 1.53
  },
  md: {
    fontSize: '$md',
    lineHeight: 1.625,
    ...capsize('-0.4489em')
  },
  lg: {
    fontSize: '$lg',
    lineHeight: 1.52,
    ...capsize('-0.3983em')
  }
}

const StyledParagraph = styled('p', {
  color: '$tonal900',
  fontFamily: '$sans',
  fontWeight: 400,
  margin: 0,
  maxWidth: '60ch',
  variants: {
    size: textVariantSize
  }
})

type TextProps = Override<
  React.ComponentPropsWithoutRef<typeof StyledParagraph>,
  {
    as?:
      | 'blockquote'
      | 'caption'
      | 'dd'
      | 'dt'
      | 'figcaption'
      | 'li'
      | 'p'
      | 'span'
    size?: 'sm' | 'md' | 'lg'
  }
>

export const Text: React.FC<TextProps> = ({
  as = 'p',
  size = 'md',
  ...rest
}) => <StyledParagraph as={as} size={size} {...rest} />

Text.displayName = 'Text'
