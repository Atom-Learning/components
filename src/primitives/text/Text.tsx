import { styled } from '../../stitches'

export const Text = styled('p', { fontFamily: 'sans', color: '$tonal900' })
