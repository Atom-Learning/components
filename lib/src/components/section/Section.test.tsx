import { render, screen } from '@testing-library/react'
import { axe } from 'jest-axe'
import * as React from 'react'
import { CheckList, CalendarEvent } from '@atom-learning/icons'

import { Text } from '../text'
import { Section, SectionItem } from './'

const Component = () => (
  <Section heading="Section Heading">
    <Section.Content>
      <Text>Section Content</Text>
    </Section.Content>
  </Section>
)

describe(`Section component`, () => {
  it('renders', async () => {
    const { container } = render(<Component />)
    await screen.findByText('Section Heading')
    await screen.findByText('Section Content')
    expect(container).toMatchSnapshot()
  })

  it('has no programmatically detectable a11y issues', async () => {
    const { container } = render(<Component />)
    expect(await axe(container)).toHaveNoViolations()
  })

  it('renders with SectionItem', async () => {
    const { container } = render(
      <Section heading="Section Heading">
        <Section.Content>
          <SectionItem
            iconBackground="$primaryDark"
            status={{ text: 'Today', theme: 'danger', icon: CalendarEvent }}
            onClick={() => console.log('click')}
          >
            <SectionItem.Icon
              is={CheckList}
              size="md"
              css={{ color: 'white' }}
            />
            <SectionItem.Heading
              as="h4"
              css={{ fontFamily: '$body' }}
              size="sm"
            >
              A small heading
            </SectionItem.Heading>
            <SectionItem.SubHeading css={{ color: '$tonal300' }} size="sm">
              A small subheading
            </SectionItem.SubHeading>
          </SectionItem>
          <SectionItem
            iconBackground="$primaryDark"
            status={{ text: 'Today', theme: 'danger', icon: CalendarEvent }}
            to="/#/"
          >
            <SectionItem.Icon
              is={CheckList}
              size="md"
              css={{ color: 'white' }}
            />
            <SectionItem.Heading
              as="h4"
              css={{ fontFamily: '$body' }}
              size="sm"
            >
              A small heading
            </SectionItem.Heading>
            <SectionItem.SubHeading css={{ color: '$tonal300' }} size="sm">
              A small subheading
            </SectionItem.SubHeading>
          </SectionItem>
          <SectionItem iconBackground="$primaryDark">
            <SectionItem.Heading
              as="h4"
              css={{ fontFamily: '$body' }}
              size="sm"
            >
              A small heading
            </SectionItem.Heading>
            <SectionItem.SubHeading css={{ color: '$tonal300' }} size="sm">
              A small subheading
            </SectionItem.SubHeading>
          </SectionItem>
        </Section.Content>
      </Section>
    )

    expect(await axe(container)).toHaveNoViolations()
    expect(container).toMatchSnapshot()
  })
})
