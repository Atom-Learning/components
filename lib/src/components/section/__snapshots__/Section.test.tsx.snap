// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section component renders 1`] = `
@media  {
  .c-dhzjXW {
    display: flex;
  }

  .c-iSnHBU {
    color: var(--colors-tonal600);
    font-family: var(--fonts-display);
    font-weight: 700;
    margin: 0;
  }

  .c-bQzyIt {
    display: grid;
  }

  .c-fIXJvv {
    color: var(--colors-tonal600);
    font-family: var(--fonts-body);
    font-weight: 400;
    margin: 0;
  }

  .c-fIXJvv > .c-fIXJvv:before,
  .c-fIXJvv > .c-fIXJvv:after {
    display: none;
  }
}

@media  {
  .c-iSnHBU-fOoUdm-size-md {
    font-size: var(--fontSizes-xl);
    line-height: 1.14;
  }

  .c-iSnHBU-fOoUdm-size-md::before {
    content: '';
    margin-bottom: -0.2174em;
    display: table;
  }

  .c-iSnHBU-fOoUdm-size-md::after {
    content: '';
    margin-top: -0.2254em;
    display: table;
  }

  .c-bQzyIt-kaVYFn-gap-2 {
    gap: var(--space-2);
  }

  .c-fIXJvv-gvmVBy-size-md {
    font-size: var(--fontSizes-md);
    line-height: 1.5;
  }

  .c-fIXJvv-gvmVBy-size-md::before {
    content: '';
    margin-bottom: -0.3864em;
    display: table;
  }

  .c-fIXJvv-gvmVBy-size-md::after {
    content: '';
    margin-top: -0.3864em;
    display: table;
  }
}

@media  {
  .c-dhzjXW-ibXmpqi-css {
    background: var(--colors-primaryLight);
    gap: var(--space-5);
    padding: calc(var(--space-4) + var(--space-2));
    flex-direction: column;
    border-radius: var(--radii-0);
  }

  .c-iSnHBU-ijkHWQd-css {
    padding-top: var(--space-2);
  }
}

<div>
  <section
    class="c-dhzjXW c-dhzjXW-ibXmpqi-css"
  >
    <h3
      class="c-iSnHBU c-iSnHBU-fOoUdm-size-md c-iSnHBU-ijkHWQd-css"
    >
      Section Heading
    </h3>
    <div
      class="c-bQzyIt c-bQzyIt-kaVYFn-gap-2 c-bQzyIt-iPJLV-css"
    >
      <p
        class="c-fIXJvv c-fIXJvv-gvmVBy-size-md"
      >
        Section Content
      </p>
    </div>
  </section>
</div>
`;

exports[`Section component renders with SectionItem 1`] = `
@media  {
  .c-dhzjXW {
    display: flex;
  }

  .c-iSnHBU {
    color: var(--colors-tonal600);
    font-family: var(--fonts-display);
    font-weight: 700;
    margin: 0;
  }

  .c-bQzyIt {
    display: grid;
  }

  .c-fIXJvv {
    color: var(--colors-tonal600);
    font-family: var(--fonts-body);
    font-weight: 400;
    margin: 0;
  }

  .c-fIXJvv > .c-fIXJvv:before,
  .c-fIXJvv > .c-fIXJvv:after {
    display: none;
  }

  .c-fkUJsw {
    align-items: center;
    background: unset;
    border: unset;
    border-radius: var(--radii-0);
    cursor: pointer;
    display: flex;
    font-family: var(--fonts-body);
    font-weight: 600;
    justify-content: center;
    padding: unset;
    text-decoration: none;
    transition: all 100ms ease-out;
    white-space: nowrap;
    width: max-content;
  }

  .c-dbrbZt {
    display: inline-block;
    fill: none;
    stroke: currentcolor;
    stroke-linecap: round;
    stroke-linejoin: round;
    vertical-align: middle;
  }

  .c-QWATl {
    justify-content: center;
    align-items: center;
    border-radius: var(--radii-0);
    border: 1px solid #FFFFFF;
    font-family: var(--fonts-body);
  }
}

@media  {
  .c-iSnHBU-fOoUdm-size-md {
    font-size: var(--fontSizes-xl);
    line-height: 1.14;
  }

  .c-iSnHBU-fOoUdm-size-md::before {
    content: '';
    margin-bottom: -0.2174em;
    display: table;
  }

  .c-iSnHBU-fOoUdm-size-md::after {
    content: '';
    margin-top: -0.2254em;
    display: table;
  }

  .c-bQzyIt-kaVYFn-gap-2 {
    gap: var(--space-2);
  }

  .c-fIXJvv-gvmVBy-size-md {
    font-size: var(--fontSizes-md);
    line-height: 1.5;
  }

  .c-fIXJvv-gvmVBy-size-md::before {
    content: '';
    margin-bottom: -0.3864em;
    display: table;
  }

  .c-fIXJvv-gvmVBy-size-md::after {
    content: '';
    margin-top: -0.3864em;
    display: table;
  }

  .c-fkUJsw-elrwsr-size-md {
    font-size: var(--fontSizes-md);
    line-height: 1.5;
    height: var(--sizes-4);
    padding-left: var(--space-5);
    padding-right: var(--space-5);
  }

  .c-dbrbZt-dMrjnF-size-md {
    height: var(--sizes-2);
    width: var(--sizes-2);
    stroke-width: 1.75;
  }

  .c-iSnHBU-dtbHih-size-sm {
    font-size: var(--fontSizes-lg);
    line-height: 1.14;
  }

  .c-iSnHBU-dtbHih-size-sm::before {
    content: '';
    margin-bottom: -0.2174em;
    display: table;
  }

  .c-iSnHBU-dtbHih-size-sm::after {
    content: '';
    margin-top: -0.2254em;
    display: table;
  }

  .c-QWATl-jtgfr-theme-danger {
    background: var(--colors-dangerLight);
    color: var(--colors-dangerDark);
  }

  .c-QWATl-eCnudN-size-xs {
    font-size: var(--fontSizes-sm);
    padding-left: var(--space-3);
    padding-right: var(--space-3);
    height: var(--sizes-2);
  }

  .c-dbrbZt-cyeZeh-size-sm {
    height: var(--sizes-1);
    width: var(--sizes-1);
    stroke-width: 1.5;
  }

  .c-fIXJvv-bndJoy-size-sm {
    font-size: var(--fontSizes-sm);
    line-height: 1.53;
  }

  .c-fIXJvv-bndJoy-size-sm::before {
    content: '';
    margin-bottom: -0.4056em;
    display: table;
  }

  .c-fIXJvv-bndJoy-size-sm::after {
    content: '';
    margin-top: -0.4056em;
    display: table;
  }

  .c-dbrbZt-fLGGKn-size-lg {
    height: var(--sizes-3);
    width: var(--sizes-3);
    stroke-width: 2;
  }
}

@media  {
  .c-fkUJsw-gsycxF-cv {
    background: var(--colors-primary);
    color: white;
  }

  .c-fkUJsw-gsycxF-cv[disabled] {
    background: var(--colors-tonal100);
    color: var(--colors-tonal400);
    cursor: not-allowed;
  }

  .c-fkUJsw-gsycxF-cv:not([disabled]):hover,
  .c-fkUJsw-gsycxF-cv:not([disabled]):focus {
    background: var(--colors-primaryMid);
    color: white;
  }

  .c-fkUJsw-gsycxF-cv:not([disabled]):active {
    background: var(--colors-primaryDark);
  }
}

@media  {
  .c-dhzjXW-ibXmpqi-css {
    background: var(--colors-primaryLight);
    gap: var(--space-5);
    padding: calc(var(--space-4) + var(--space-2));
    flex-direction: column;
    border-radius: var(--radii-0);
  }

  .c-iSnHBU-ijkHWQd-css {
    padding-top: var(--space-2);
  }

  .c-fkUJsw-iiBxIJp-css {
    height: unset;
    width: unset;
    display: flex;
    gap: var(--space-4);
    background: white;
    border-radius: var(--radii-2);
    padding: var(--space-4);
    justify-content: flex-start;
    align-items: center;
    text-align: left;
  }

  .c-fkUJsw-iiBxIJp-css::before,
  .c-fkUJsw-iiBxIJp-css::after {
    display: none !important;
  }

  .c-fkUJsw-iiBxIJp-css:hover:not([disabled]),
  .c-fkUJsw-iiBxIJp-css:active:not([disabled]),
  .c-fkUJsw-iiBxIJp-css:focus:not([disabled]) {
    background: white;
    text-decoration: none;
  }

  .c-fkUJsw-iiBxIJp-css:hover:not([disabled]) > svg,
  .c-fkUJsw-iiBxIJp-css:active:not([disabled]) > svg,
  .c-fkUJsw-iiBxIJp-css:focus:not([disabled]) > svg {
    animation: k-cBFJMA 250ms ease;
    animation-fill-mode: forwards;
  }

  .c-fkUJsw-iiBxIJp-css > svg {
    animation: k-hKVLpM 250ms ease;
    animation-fill-mode: forwards;
  }

  .c-dhzjXW-iehXZNA-css {
    display: none;
    align-items: center;
    justify-content: center;
    background: var(--colors-primaryDark);
    height: var(--sizes-5);
    width: var(--sizes-5);
    border-radius: var(--radii-round);
  }

@media (min-width: 550px) {
    .c-dhzjXW-iehXZNA-css {
      display: flex;
    }
}

  .c-dbrbZt-ifjEkWJ-css {
    color: white;
  }

  .c-dhzjXW-icdAfUY-css {
    gap: var(--space-3);
    flex-direction: column;
    align-items: flex-start;
  }

@media (min-width: 550px) {
    .c-dhzjXW-icdAfUY-css {
      gap: var(--space-1);
    }
}

  .c-dhzjXW-ikdofoX-css {
    gap: var(--space-2);
  }

  .c-iSnHBU-icyRcZm-css {
    font-family: var(--fonts-body);
  }

  .c-dhzjXW-ihhNUSJ-css {
    padding-left: var(--space-1);
    padding-right: var(--space-1);
    border-radius: var(--radii-1);
  }

  .c-PJLV-iguVXTk-css {
    white-space: nowrap;
    overflow-x: hidden;
    text-overflow: ellipsis;
    padding-top: var(--space-0);
    padding-bottom: var(--space-0);
  }

  .c-dhzjXW-ieVEbtG-css {
    gap: var(--space-1);
    align-items: center;
  }

  .c-fIXJvv-iesazQZ-css {
    color: var(--colors-tonal300);
  }

  .c-dbrbZt-ifAkqkn-css {
    margin-left: auto;
    color: var(--colors-primary);
    display: none;
  }

@media (min-width: 550px) {
    .c-dbrbZt-ifAkqkn-css {
      display: block;
    }
}
}

<div>
  <section
    class="c-dhzjXW c-dhzjXW-ibXmpqi-css"
  >
    <h3
      class="c-iSnHBU c-iSnHBU-fOoUdm-size-md c-iSnHBU-ijkHWQd-css"
    >
      Section Heading
    </h3>
    <div
      class="c-bQzyIt c-bQzyIt-kaVYFn-gap-2 c-bQzyIt-iPJLV-css"
    >
      <button
        class="c-fkUJsw c-fkUJsw-elrwsr-size-md c-fkUJsw-gsycxF-cv c-fkUJsw-iiBxIJp-css"
        type="button"
      >
        <div
          class="c-dhzjXW c-dhzjXW-iehXZNA-css"
        >
          <svg
            aria-hidden="true"
            class="c-dbrbZt c-dbrbZt-dMrjnF-size-md c-dbrbZt-ifjEkWJ-css"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 6h10m-10 6h10m-10 6h10M3.5 11.5l1 1 2-2m-3-5l1 1 2-2"
            />
            <circle
              cx="5"
              cy="18"
              r="0.75"
            />
          </svg>
        </div>
        <div
          class="c-dhzjXW c-dhzjXW-icdAfUY-css"
        >
          <div
            class="c-dhzjXW c-dhzjXW-ikdofoX-css"
          >
            <h4
              class="c-iSnHBU c-iSnHBU-dtbHih-size-sm c-iSnHBU-icyRcZm-css"
            >
              A small heading
            </h4>
            <div
              class="c-dhzjXW c-QWATl c-QWATl-jtgfr-theme-danger c-QWATl-eCnudN-size-xs c-dhzjXW-ihhNUSJ-css"
              role="status"
            >
              <div
                class="c-PJLV c-PJLV-iguVXTk-css"
              >
                <div
                  class="c-dhzjXW c-dhzjXW-ieVEbtG-css"
                >
                  <svg
                    aria-hidden="true"
                    class="c-dbrbZt c-dbrbZt-cyeZeh-size-sm"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M3 5h18v16H3V5zm18 4H3m4-4V3m10 2V3"
                    />
                    <rect
                      height="2"
                      width="2"
                      x="15"
                      y="15"
                    />
                  </svg>
                  Today
                </div>
              </div>
            </div>
          </div>
          <p
            class="c-fIXJvv c-fIXJvv-bndJoy-size-sm c-fIXJvv-iesazQZ-css"
          >
            A small subheading
          </p>
        </div>
        <svg
          aria-hidden="true"
          class="c-dbrbZt c-dbrbZt-fLGGKn-size-lg c-dbrbZt-ifAkqkn-css"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 18l6-6-6-6M3 12h17"
          />
          <path
            d="M21 12h-1"
            stroke-linecap="round"
          />
        </svg>
      </button>
      <a
        class="c-fkUJsw c-fkUJsw-elrwsr-size-md c-fkUJsw-gsycxF-cv c-fkUJsw-iiBxIJp-css"
        href="/#/"
      >
        <div
          class="c-dhzjXW c-dhzjXW-iehXZNA-css"
        >
          <svg
            aria-hidden="true"
            class="c-dbrbZt c-dbrbZt-dMrjnF-size-md c-dbrbZt-ifjEkWJ-css"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 6h10m-10 6h10m-10 6h10M3.5 11.5l1 1 2-2m-3-5l1 1 2-2"
            />
            <circle
              cx="5"
              cy="18"
              r="0.75"
            />
          </svg>
        </div>
        <div
          class="c-dhzjXW c-dhzjXW-icdAfUY-css"
        >
          <div
            class="c-dhzjXW c-dhzjXW-ikdofoX-css"
          >
            <h4
              class="c-iSnHBU c-iSnHBU-dtbHih-size-sm c-iSnHBU-icyRcZm-css"
            >
              A small heading
            </h4>
            <div
              class="c-dhzjXW c-QWATl c-QWATl-jtgfr-theme-danger c-QWATl-eCnudN-size-xs c-dhzjXW-ihhNUSJ-css"
              role="status"
            >
              <div
                class="c-PJLV c-PJLV-iguVXTk-css"
              >
                <div
                  class="c-dhzjXW c-dhzjXW-ieVEbtG-css"
                >
                  <svg
                    aria-hidden="true"
                    class="c-dbrbZt c-dbrbZt-cyeZeh-size-sm"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M3 5h18v16H3V5zm18 4H3m4-4V3m10 2V3"
                    />
                    <rect
                      height="2"
                      width="2"
                      x="15"
                      y="15"
                    />
                  </svg>
                  Today
                </div>
              </div>
            </div>
          </div>
          <p
            class="c-fIXJvv c-fIXJvv-bndJoy-size-sm c-fIXJvv-iesazQZ-css"
          >
            A small subheading
          </p>
        </div>
        <svg
          aria-hidden="true"
          class="c-dbrbZt c-dbrbZt-fLGGKn-size-lg c-dbrbZt-ifAkqkn-css"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 18l6-6-6-6M3 12h17"
          />
          <path
            d="M21 12h-1"
            stroke-linecap="round"
          />
        </svg>
      </a>
      <div
        class="c-fkUJsw c-fkUJsw-elrwsr-size-md c-fkUJsw-gsycxF-cv c-fkUJsw-iiBxIJp-css"
        type="button"
      >
        <div
          class="c-dhzjXW c-dhzjXW-icdAfUY-css"
        >
          <div
            class="c-dhzjXW c-dhzjXW-ikdofoX-css"
          >
            <h4
              class="c-iSnHBU c-iSnHBU-dtbHih-size-sm c-iSnHBU-icyRcZm-css"
            >
              A small heading
            </h4>
          </div>
          <p
            class="c-fIXJvv c-fIXJvv-bndJoy-size-sm c-fIXJvv-iesazQZ-css"
          >
            A small subheading
          </p>
        </div>
      </div>
    </div>
  </section>
</div>
`;
